src/encoder/external_encoder.ml

```diff
@@ -88,15 +88,15 @@ let encoder id ext =
       let len = puller bytes 0 Utils.pagesize in
       match len with
         | 0 when !is_stop -> Condition.signal condition
-        | _ -> buf := Strings.add_subbytes !buf bytes 0 len
+        | _ -> Strings.Mutable.unsafe_add_subbytes buf bytes 0 len
     end;
     `Continue)
   in
 
-  let flush_buffer = Tutils.mutexify mutex (fun () ->
-      let ans = !buf in
-      buf := Strings.empty;
-      ans)
+  let flush_buffer () = 
+    let ans = Strings.Mutable.to_strings buf in
+    Strings.Mutable.flush buf;
```

Comment:
I think that is would be more natural to have `flush` both empty the buffer *and* return the previous contents, so that this would be only one line.

Reply:
Good idea.